src/lambertw_omega.jl

```diff
@@ -0,0 +1,32 @@
+# lazy-initialized LambertW Omega at 256-bit precision
+const LambertW_Omega_BigFloat256 = Ref{BigFloat}()
+
+# compute BigFloat Omega constant at arbitrary precision
+function compute_LambertW_Omega()
+    # initialize Omega_BigFloat256
+    isassigned(LambertW_Omega_BigFloat256) ||
+        (LambertW_Omega_BigFloat256[] = BigFloat("0.5671432904097838729999686622103555497538157871865125081351310792230457930866845666932194"))
+    o = LambertW_Omega_BigFloat256[] # initial value
+    precision(BigFloat) <= 256 && return o
+    # iteratively improve the precision of the constant
+    myeps = eps(BigFloat)
+    for _ in 1:100
+        o_ = (1 + o) / (1 + exp(o))
+        abs(o - o_) <= myeps && break
+        o = o_
+    end
+    return o
+end
```

Comment:
I'm not sure if it's worth caching the value in `LambertW_Omega_BigFloat256`. I don't think this is done for other constants? Also it seems it would return a value of a different precision than requested in the case `precision(BigFloat) < 256`.

Reply:
The other constants either have corresponding `mpfr_const_xxx()` functions or simple formulas.
But I agree it would be more straightforward to have just
```julia
    o = BigFloat("0.5671432904097838729999686622103555497538157871865125081351310792230457930866845666932194") # initial value with 256-bit precision
    precision(BigFloat) <= 256 && return o
```
if `BigFloat("...")` is fast enough.

Reply:
Now the returned *BigFloat* constant should respect the current precision better, and the appropriate tests were added.
I still have kept the `lambertw_Omega_BigFloat256` as I assume it's faster to reuse the initialized *BigFloat* than parse it each time.

Reply:
This is much slower than the code in LambertW.jl. The two should be benchmarked explicitly before a PR would be accepted.

 But, I agree with the others that it does not make sense to move the omega constant out of LambertW.jl. It should stay in LambertW.jl and the latter should be moved into SpecialFuncions